src/mtype.h

```diff
@@ -268,8 +271,15 @@ class Type : public RootObject
     virtual bool isscope();
     virtual bool isString();
     virtual bool isAssignable();
+
+    virtual TypeStruct* isTypeStruct() { return NULL; }
+    virtual TypeClass* isTypeClass() { return NULL; }
+    virtual bool isTypeAggregate() { return this->isTypeStruct() || this->isTypeClass(); }
+    virtual TypeDArray* isTypeDArray() { return NULL; }
```

Comment:
I like the inline style on if-statement, but I'll never wish adding such `isTypeXXX()` functions.

Because it doesn't increase code readability. Today, all dmd developers should learn the relation between `Txxx` enums and `TypeXXX` classes. So such the functions will just increase the code style confusion.